### ISSUE TITLE
Problem with LVGL subpixel rendering with inverted RGB

### ISSUE THREAD

**Issue by EnricoLusaEGF** [kisvegabor/lvgl#3764]
### Describe the bug

I’m working on a project that uses LVGL 8.3.2.

I’m using fonts created with the LVGL converter, with BPP=4 and subpixel rendering.

The display we are using is “upside down”, so the pixels are in BGR order from the user point of view.

To make subpixel rendering work in this configuration I’ve set LV_DRAW_SW_FONT_SUBPX_BGR to 1.

This has the desired effect: the font is sharper and under a lens it’s visible that the curves are correct.

The problem is that now when I set the font to a color different than black, the colors are skewed

![image](https://user-images.githubusercontent.com/78972584/197518403-163dfe09-9566-42e5-b22c-60dc24bbb222.png) vs ![image](https://user-images.githubusercontent.com/78972584/197518470-2b86b1a2-ff86-45dd-9b58-c1c3bd997bea.png)
_with BGR  vs  original_

(this screenshot is taken from the vscode simulator, where the effect is identical to the one on my hardware).

(The color depth doesn't affect this).

While writing this report I started fixing it and I made some progress, I will leave this here for information and to refer this in the pull request if everything works out.

### To Reproduce

You can reproduce this by simply running the simulator with the widgets demo, setting 
LV_DRAW_SW_FONT_SUBPX_BGR = 1
in lv_conf.h
and replacing some of the default fonts (in lv_demo_widgets.c) with lv_font_montserrat_12_subpx

![image](https://user-images.githubusercontent.com/78972584/197520149-f14b53bc-d5d9-48e5-81a6-9fedbe46b0bc.png)

### A bit more analysis

I think the problem comes from the fact that with LV_DRAW_SW_FONT_SUBPX_BGR we simply invert B and R.
This is ok with black because we basically either have black pixel or boundary pixel. Inverted black is still black, and the boundary correctly mirror the subpixel.
For example(you can see the inverted colors of the borders):
![image](https://user-images.githubusercontent.com/78972584/197522534-56b53484-945a-4957-821d-0dcd44f6c5a5.png)
This on a real screen (inverted) behave like this:
**(wrong RGB order)**
![image](https://user-images.githubusercontent.com/78972584/197523901-abff6249-8524-4645-9d6c-71b207a174d8.png)
**(correct  BGR order)**
![image](https://user-images.githubusercontent.com/78972584/197523791-eecce893-32cb-4dc6-bd4f-71c65d8d1872.png)

When the text is colored tho, inverting all B and R will change all the colors, not only the boundaries:
![image](https://user-images.githubusercontent.com/78972584/197538501-f3748def-bd5a-45e0-ae13-e5fea5a71052.png)

This is even a bigger issue when the background isn't white, because we are blending normal colors with "inverted" colors:
![image](https://user-images.githubusercontent.com/78972584/197553094-bec21a9f-125a-4019-abe2-c8cda7cf8c1a.png)

**Comment by kisvegabor** [comment]
Hi,

Thanks for reporting it. It's a big surprise that this issues was not spotted so far. 


Based on your hints I've looked into it. I'm not entirely sure this might solve the issue: please change these lines
https://github.com/lvgl/lvgl/blob/903e94b716ca1b32cdb51de11df679953699e53b/src/draw/sw/lv_draw_sw_letter.c#L491-L497

to
```c
#if LV_FONT_SUBPX_BGR
                res_color.ch.red = (uint32_t)((uint16_t)txt_rgb[0] * font_rgb[2] + (bg_rgb[0] * (255 - font_rgb[2]))) >> 8;
                res_color.ch.blue = (uint32_t)((uint16_t)txt_rgb[2] * font_rgb[0] + (bg_rgb[2] * (255 - font_rgb[0]))) >> 8;
#else
                res_color.ch.red = (uint32_t)((uint16_t)txt_rgb[0] * font_rgb[0] + (bg_rgb[0] * (255 - font_rgb[0]))) >> 8;
                res_color.ch.blue = (uint32_t)((uint16_t)txt_rgb[2] * font_rgb[2] + (bg_rgb[2] * (255 - font_rgb[2]))) >> 8;
#endif
```

**Comment by EnricoLusaEGF** [comment]
Nice! 
That was easier than I thought, I was doing a lookahead to detect boundaries :)

I've tested it on my hardware and it seems to be working as expected. I'll try with different BPP just to be sure.

**Comment by EnricoLusaEGF** [comment]
Ok tested with all BPPs, colored text, colored background on BGR screen, everything looks good.

Thanks!

**Comment by kisvegabor** [comment]
Great! Thanks for testing it.

I've pushed the fix to both `release/v8.3` and `master`.
